main.go

```diff
@@ -114,7 +114,7 @@ func main() {
 	case "aws":
 		p, err = provider.NewAWSProvider(cfg.DomainFilter, cfg.DryRun)
 	case "inmemory":
-		p, err = provider.NewInMemoryProviderWithDomainAndLogging("example.com"), nil
+		p, err = provider.NewInMemoryProvider(provider.InMemoryWithDomain(cfg.DomainFilter), provider.InMemoryWithLogging()), nil
```

Comment:
What's the benefit of doing it? Seems to me that this replaces a specific list of parameters or a single struct with a list of `opts` functions.